packages/site/src/components/Buttons.tsx

```diff
@@ -102,6 +113,10 @@ export const HeaderButtons = ({
     return <ConnectButton onClick={onConnectClick} />;
   }
 
+  if (state.snap && isLocalSnap(state.snap?.id)) {
```

Comment:
You already have `shouldDisplayReconnectButton()` in Home.tsx. Maybe extract it and use in both places?